homeassistant/helpers/frame.py

```diff
@@ -99,10 +100,14 @@ def report(
         msg = f"Detected code that {what}. Please report this issue."
         if error_if_core:
             raise RuntimeError(msg) from err
-        _LOGGER.warning(msg, stack_info=True)
+        if not log_custom_component_only:
+            _LOGGER.warning(msg, stack_info=True)
         return
 
-    _report_integration(what, integration_frame, level)
+    if not log_custom_component_only or (
+        integration_frame.custom_integration and log_custom_component_only
```

Comment:
We don't need to check the second condition after the `and` since we already checked it before and we're after an `or` here.